minio/api.py

```diff
@@ -860,14 +872,28 @@ def put_object(self, bucket_name, object_name, data, length,
         metadata['Content-Type'] = 'application/octet-stream' if \
             not content_type else content_type
 
-        if length > part_size:
-            return self._stream_put_object(bucket_name, object_name,
-                                           data, length, metadata=metadata,
-                                           sse=sse, progress=progress,
-                                           part_size=part_size)
+        current_data = []
+        iterable = iter(read_complete_file_in_chunks(data, part_size))
+        stream = False
+        try:
+            current_data.append(next(iterable))
+            stream = True
```

Comment:
shouldn't the stream be `False` if the `len(current_data) == 1`

Reply:
current_data has alway a size of 1. I used a list to use `itertools.chain` later(line 888). An alternative woul be:
```
           current_data = None
           iterable = iter(read_complete_file_in_chunks(data, part_size))
           stream = False
           try:
               current_data = next(iterable)
               stream = True
           except StopIteration:
               pass # we catch the StopIteration to know if there are more data than part_size or not
           
           ...
           
           itertools.chain([current_data], iterable), # line 888 we want to chain the already read data with the rest of the chunks
           ...
```

Reply:
@chickeaterbanana what i meant was - if `iterable` yields only one element (for a smaller object - single put), `stream = True` will be true always and we would try multipart upload for those objects.

And stream will be set to False only when next(iterable) throws `StopIteration` for the first iteration itself

Reply:
adding `current_data.append(next(iterable))` after `stream = False` would solve this i supppose. Meaning, if second iteration of next(iterable) did not throw `StopIteration`, i can be identified as object stream (stream = True)

Reply:
You are correct. To check this I have added a unittest. I corrected it.